example/upgrade_server.rb

```diff
@@ -99,16 +110,19 @@ def on_message_complete
   end
 
   conn.on(:stream) do |stream|
+
     log = Logger.new(stream.id)
-    req, buffer = {}, ''
+    request_headers = {}
+    req = request_header_hash(request_headers)
```

Comment:
Do we actually need both `req` and `request_headers`? Can we simplify it by removing one of them? Also we're passing in an empty value which is also the default on the method..